src/rpc.rs

```diff
@@ -32,91 +29,77 @@ impl RpcClient for UPClientZenoh {
         options: CallOptions,
     ) -> RpcClientResult {
         // Validate UUri
-        UriValidator::validate(&topic)
-            .map_err(|_| RpcMapperError::UnexpectedError(String::from("Wrong UUri")))?;
+        UriValidator::validate(&topic).map_err(|_| {
+            let msg = "Invalid UUri for invoke_method".to_string();
+            log::error!("{msg}");
+            RpcMapperError::UnexpectedError(msg)
+        })?;
 
         // Get Zenoh key
         let Ok(zenoh_key) = UPClientZenoh::to_zenoh_key_string(&topic) else {
-            return Err(RpcMapperError::UnexpectedError(String::from(
-                "Unable to transform to Zenoh key",
-            )));
+            let msg = "Unable to transform UUri to Zenoh key".to_string();
+            log::error!("{msg}");
+            return Err(RpcMapperError::UnexpectedError(msg));
         };
 
-        // Get the data from UPayload
-        let Some(Data::Value(buf)) = payload.data else {
-            // Assume we only have Value here, no reference for shared memory
-            return Err(RpcMapperError::InvalidPayload(String::from(
-                "The data in UPayload should be Data::Value",
-            )));
-        };
-
-        // Generate UAttributes
+        // Create UAttributes and put into Zenoh user attachment
         let uuid_builder = UUIDBuilder::new();
-        let reqid = UUIDBuilder::new().build();
-        // Create response address
-        let mut source = topic.clone();
-        source.resource = Some(UResourceBuilder::for_rpc_response()).into();
-        // Create UMessage
-        let umessage = if let Some(token) = options.token() {
-            UMessageBuilder::request(&topic, &source, &reqid, options.timeout())
-                .with_token(&token.to_string())
-                .build(&uuid_builder)
-        } else {
-            UMessageBuilder::request(&topic, &source, &reqid, options.timeout())
-                .build(&uuid_builder)
-        };
-        // Extract uAttributes
-        let Ok(UMessage {
-            attributes: uattributes,
-            ..
-        }) = umessage
-        else {
-            return Err(RpcMapperError::UnexpectedError(String::from(
-                "Unable to create uAttributes",
-            )));
-        };
+        let mut uattributes = UAttributes::request(
```

Comment:
Using the function @sophokles73 added :rocket: :rocket: :rocket: :rocket: :rocket: